src/fobnail-attester.c

```diff
@@ -20,6 +21,31 @@ static volatile sig_atomic_t quit = 0;
 static const char LISTEN_ADDRESS[] = "0.0.0.0";
 static unsigned int port = COAP_DEFAULT_PORT; /* default port 5683 */
 
+// Extract nonce from given CBOR blob. Returned nonce is valid as long CBOR blob
+// is valid.
+static UsefulBufC get_nonce(UsefulBufC data) {
+    QCBORError uErr;
+    QCBORDecodeContext ctx;
+    UsefulBufC nonce;
+
+    if (UsefulBuf_IsNULLOrEmptyC(data))
+        return NULLUsefulBufC;
+
+    QCBORDecode_Init(&ctx, data, QCBOR_DECODE_MODE_NORMAL);
+    QCBORDecode_EnterMap(&ctx, NULL);
+        QCBORDecode_GetByteStringInMapSZ(&ctx, "nonce", &nonce);
+    QCBORDecode_ExitMap(&ctx);
```

Comment:
@arturkow2000 have you tested if this works with CBOR maps that also have other fields than just `nonce`?

Reply:
Just tested by adding few fields, still works.